Plugin/Sales/Model/ResourceModel/Order/Creditmemo/Relation/Refund.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * ClassyLlama_AvaTax
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @copyright  Copyright (c) 2016 Avalara, Inc.
+ * @license    http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+
+namespace ClassyLlama\AvaTax\Plugin\Sales\Model\ResourceModel\Order\Creditmemo\Relation;
+
+use Magento\Framework\Model\AbstractModel;
+use Magento\Sales\Api\Data\CreditmemoInterface;
+
+/**
+ * Class Refund
+ *
+ * Plugin for \Magento\Sales\Model\ResourceModel\Order\Creditmemo\Relation\Refund
+ */
+class Refund
+{
+    /**
+     * @param \Magento\Sales\Model\ResourceModel\Order\Creditmemo\Relation\Refund $subject
+     * @param \Closure $proceed
+     * @param AbstractModel $object
+     * @return void
+     * @throws \Exception
+     */
+    public function aroundProcessRelation(
+        \Magento\Sales\Model\ResourceModel\Order\Creditmemo\Relation\Refund $subject,
+        \Closure $proceed,
+        AbstractModel $object
+    ) {
+        /*
+         * The \Magento\Framework\Model\ResourceModel\Db\AbstractDb save() method makes a call to
+         * processRelations() on the credit memo ResourceModel when the credit memo object is saved
+         * $this->objectRelationProcessor->validateDataIntegrity($this->getMainTable(), $object->getData());
+         */
+
+        $avataxDataChanges = false;
+        $prepareDataChanges = false;
+
+        // Check to see if object is a credit memo
+        if ($object instanceof CreditmemoInterface) {
+
+            // Check to see if dataHasChangedFor AvaTax specific fields
+            if (
+                $object->dataHasChangedFor('avatax_is_unbalanced') ||
```

Comment:
If you do the refactor like I suggest below, you'll likely want to refactor this to loop over the `$avaTaxAttributes` array.